apps/remix-ide/src/app/tabs/compileTab/compileTab.js

```diff
@@ -78,8 +78,30 @@ class CompileTab {
     })
   }
 
-  runCompiler () {
+  async isHardhatProject () {
+    if (this.fileManager.mode === 'localhost') {
+      return await this.fileManager.exists('hardhat.config.js')
+    } else return false
+  }
+
+  runCompiler (hhCompilation) {
     try {
+      if (this.fileManager.mode === 'localhost' && hhCompilation) {
+        const { currentVersion, optimize, runs } = this.compiler.state
+        const fileContent = `module.exports = {
+          solidity: '${currentVersion.substring(0, currentVersion.indexOf('+commit'))}',
+          settings: {
+            optimizer: {
+              enabled: ${optimize},
+              runs: ${runs}
+            }
+          }
+        }
+        `
+        const configFilePath = 'remix-compiler.config.js'
+        this.fileManager.setFileContent(configFilePath, fileContent)
+        this.fileManager.appManager.call('hardhat', 'compile', configFilePath)
```

Comment:
you could just do `fileManager.call('hardhat', 'compile', configFilePath)` or better to turn this component as a plugin.

Reply:
updated